Pocket/pocket-ios
# feat(search): present view for errors

## Summary
Present view for errors from online search. 

## References 
IN-1110

## Implementation Details
Added a new `EmptyStateViewModel` called `ErrorEmptyState` to handle errors returned from our Online Search for Archive and All Items. Added new `bannerData` to our `SearchViewModel` which is presented when we set `showBanner` to true when an Online Search for Saves fails. This banner is presented for offline state and unhandled error states.

## Test Steps
For saves search:
- [ ] When I submit a search query 
AND the query fails  
AND the failure is an unhandled error
THEN perform local search
AND display banner notifying the user something went wrong

For archive search:
- [ ] When I submit a search query 
AND the query fails  
AND the failure is an unhandled error
THEN present an error empty state to communicate the error

For all items search:
- [ ] When I submit a search query 
AND the query fails  
AND the failure is an unhandled error
THEN present an error empty state to communicate the error
Designs: See gig attached. 

## PR Checklist:

## Screenshots

|Archive / All Items Error State|Saves Error State|
|------|-----|
|![Archive / All Items Error State](https://user-images.githubusercontent.com/6743397/218778470-7122c35f-ea7e-4718-b58c-4c8923788655.png)|![Saves Error State](https://user-images.githubusercontent.com/6743397/218778473-5615fe82-3313-4409-9e62-5fc1e2a46607.png)|
